pkg/openfeature/provider.go

```diff
@@ -16,10 +16,10 @@ const (
 // vendors should implement
 type FeatureProvider interface {
 	Metadata() Metadata
-	BooleanEvaluation(flag string, defaultValue bool, evalCtx map[string]interface{}) BoolResolutionDetail
-	StringEvaluation(flag string, defaultValue string, evalCtx map[string]interface{}) StringResolutionDetail
-	FloatEvaluation(flag string, defaultValue float64, evalCtx map[string]interface{}) FloatResolutionDetail
-	IntEvaluation(flag string, defaultValue int64, evalCtx map[string]interface{}) IntResolutionDetail
+	BooleanEvaluation(flag string, defaultValue bool, evalCtx map[string]interface{}) ResolutionDetail
+	StringEvaluation(flag string, defaultValue string, evalCtx map[string]interface{}) ResolutionDetail
+	FloatEvaluation(flag string, defaultValue float64, evalCtx map[string]interface{}) ResolutionDetail
+	IntEvaluation(flag string, defaultValue int64, evalCtx map[string]interface{}) ResolutionDetail
```

Comment:
It was nice having the typed ResolutionDetail where the `Value` was typed. That way, when you called `getBooleanValue(...)` (or getBooleanDetail) you got a `boolean` value back, rather than a generic `interface{}`. It would be nice if we could keep this.

Wolud a possible fix be to make `ResolutionDetail` abstract (if that's even a thing in go) and remove the `Value: interface{}` from there, and keep the typed version in BoolResolutionValue etc?

Reply:
Good suggestion, done :)